src/features/views/components/ViewColumnDialog/SurveyResponseConfig.tsx

```diff
@@ -0,0 +1,226 @@
+          };
+        });
+      } else {
+        const optionTitle = selectedQuestion.question.options?.find(
+          (option) => option.id === (selectedOption as unknown as number)
+        );
+        if (optionTitle === undefined) {
+          return [];
+        }
+        return [
+          {
+            config: {
+              option_id: selectedOption as unknown as number,
+              survey_id: selectedQuestion.id,
+            },
+            title: optionTitle.text,
+            type: COLUMN_TYPE.SURVEY_OPTION,
+          },
+        ];
+      }
+    }
+  };
+
+  return (
+    <ZUIQuery queries={{ surveysQuery }}>
+      {({ queries: { surveysQuery: successSurveysQuery } }) => {
+        const selectedSurvey = successSurveysQuery.data.find(
+          (survey) => survey.id == surveyId
+        );
+        const questionFromSurvey = selectedSurvey?.elements.filter(
+          (elem) => elem.type == ELEMENT_TYPE.QUESTION
+        );
+
+        return (
+          <FormControl sx={{ width: 300 }}>
+            <TextField
+              fullWidth
+              label={intl.formatMessage({
+                id: 'misc.views.columnDialog.choices.surveyResponse.surveyField',
+              })}
+              margin="normal"
+              onChange={onSurveyChange}
+              select
+              value={surveyId || ''}
+              variant="standard"
+            >
+              {successSurveysQuery.data.map((survey) => (
+                <MenuItem key={survey.id} value={survey.id}>
+                  {survey.title}
+                </MenuItem>
+              ))}
+            </TextField>
+            {surveyId ? (
+              <Autocomplete
+                disabled={!surveyId}
+                fullWidth
+                getOptionLabel={(option) => option.question.question}
+                onChange={(evt, value) => {
+                  if (value !== null) {
+                    setSelectedQuestion(value);
+                    const columns = makeColumnsText(value);
+                    if (columns !== undefined) {
+                      onOutputConfigured(columns);
+                    }
+                  }
+                }}
+                options={questionFromSurvey || []}
+                renderInput={(params) => (
+                  <TextField
+                    {...params}
+                    inputProps={{
+                      ...params.inputProps,
+                    }}
+                    label={intl.formatMessage({
+                      id: 'misc.views.columnDialog.choices.surveyResponse.questionField',
+                    })}
+                    variant="standard"
+                  ></TextField>
+                )}
+                value={selectedQuestion}
+              />
+            ) : (
+              ''
+            )}
+            {surveyId &&
+            selectedQuestion &&
+            selectedQuestion?.question.response_type === RESPONSE_TYPE.TEXT ? (
```

Comment:
I don't like complicated ternaries like this in JSX. They are hard to follow, and it's difficult to see where they start and end. It's made even more difficult by the _nested_ ternary here (branches again on line 183).

If the logic isn't really nested, but just a way to make a three-way conditional, I would use the `condition && <Component/>` pattern three times instead.